lib/centralisation/providers/favoritesUtils.dart

```diff
@@ -0,0 +1,9 @@
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:myecl/centralisation/class/module.dart';
+import 'dart:convert';
+
+Future<void> saveFavoritesToSharedPreferences(List<Module> favoritesList) async {
+  final prefs = await SharedPreferences.getInstance();
+  final favoritesJson = favoritesList.map((module) => module.toJson()).toList();
+  await prefs.setString('favorites', json.encode(favoritesJson));
+}
```

Comment:
You have created a util to write to storage, but the function to read from it is placed in the provider, please place both of them in the same file, and this one is really good for that